docs/tutorial/pytorch_training.md

```diff
@@ -0,0 +1,175 @@
+with elasticity using DLRover. Users only need to make some simple changes
+of native PyTorch training codes. We have provided the
+[CNN example](../../model_zoo/pytorch/mnist_cnn.py) to show how to
+train a CNN model with the MNIST dataset.
+
+## Develop a Torch Model with DLRover. 
+
+### Setup the Environment Using ElasticTrainer
+
+Users need to set up the environment by `ElasticTrainer`. The `ElasticTrainer`
+will set the IP of rank-0 into `MASTER_ADDR`, because the node with rank-0
+will change during elasticity.
+
+```python
+from dlrover.trainer.torch.elastic import ElasticTrainer
+
+ElasticTrainer.setup()
+```
+
+### Develop the ElasticDataset.
+
+Firstly, users need to write the path of the sample into a `Text` file.
+The path can be a location path of a file or a linke to download
+the sample data from a remote storage. For example, we can create a Text
+file to storage the location path and lable of MNIST dataset.
+
+```text
+/data/mnist_png/training/9/37211.png,9
+/data/mnist_png/training/9/51194.png,9
+/data/mnist_png/training/9/374.png,9
+/data/mnist_png/training/1/29669.png,1
+/data/mnist_png/training/1/19782.png,1
+/data/mnist_png/training/1/42786.png,1
+/data/mnist_png/training/8/41017.png,8
+/data/mnist_png/training/8/13037.png,8
+/data/mnist_png/training/8/7101.png,8
+```
+
+Then, we can develop a dataset to support elastic training
+using the Text file.
+
+```python
+from dlrover.trainer.torch.elastic_dataset import ElasticDataset
+
+
+class ElasticMnistDataset(ElasticDataset):
+    def __init__(self, path, batch_size, epochs, shuffle, checkpoint_path):
+        """The dataset supports elastic training.
+
+        Args:
+            path: str, the path of dataset meta file. For example, if the image
+                is stored in a folder. The meta file should be a
+                text file where each line is the absolute path of a image.
+            batch_size: int, the size of batch samples to compute gradients
+                in a trainer process.
+            epochs: int, the number of epoch.
+            shuffle: bool, whether to shuffle samples in the dataset.
+            checkpoint_path: the path to save the checkpoint of shards
+                int the dataset.
+        """
+        super(ElasticMnistDataset, self).__init__(
+            path,
+            batch_size,
+            epochs,
+            shuffle,
+            checkpoint_path,
+        )
+
+    def read_sample(self, index):
+        """
+        Read the sample by the index. Users can get the location
+        path by the index from the text file and write codes
+        to read sample data.
+        """
+        pass
+```
+
+### Wrap the Training Step using ElasticTrainer
+
+To keep the total batch size fixed during elastic training,
+users need to create an `ElasticTrainer` to wrap the model, optimizer
+and scheduler. `ElasticTrainer` can keep the total batch size
+fixed by accumulating gradients if the number of worker decreases.
```

Comment:
It'd better also explain how to fix the batch size when the number of workers increases here.